### ISSUE TITLE
Linking error for v1.0.0 and libprotoc 3.1.0

### ISSUE THREAD

**Issue by ojmakke** [grpc/grpc#9459]
Hello, 

I successfully installed"protoc", and I am using libprotoc 3.1.0

I followed the instructions to install grpc v1.0.0. I get the following linker error (using gcc 4.9.2 on Debian Jessie)

_[HOSTLD]  Linking /grpc/bins/opt/grpc_cpp_plugin
/grpc/objs/opt/src/compiler/cpp_plugin.o: In function `std::vector<int, std::allocator<int> >::push_back(int const&)':
/usr/include/c++/4.9/bits/stl_vector.h:923: undefined reference to `google::protobuf::FileDescriptorProto::kSyntaxFieldNumber'
/usr/include/c++/4.9/bits/stl_vector.h:923: undefined reference to `google::protobuf::FileDescriptorProto::kSyntaxFieldNumber'
/grpc/objs/opt/src/compiler/cpp_plugin.o:(.data.rel.ro._ZTV16CppGrpcGenerator[_ZTV16CppGrpcGenerator]+0x28): undefined reference to `google::protobuf::compiler::CodeGenerator::GenerateAll(std::vector<google::protobuf::FileDescriptor const*, std::allocator<google::protobuf::FileDescriptor const*> > const&, std::string const&, google::protobuf::compiler::GeneratorContext*, std::string*) const'
collect2: error: ld returned 1 exit status
Makefile:11164: recipe for target '/grpc/bins/opt/grpc_cpp_plugin' failed_

I am suspecting some version mismatch somewhere, but I could not find information on what version goes with what.

**Comment by ncteisen** [comment]
How did you install protoc? Have you installed the protobuf lib onto your system as well? That would look something like
```
cd third_party/protobuf
./autogen.sh
./configure
make
make check
sudo make install
```

**Comment by ojmakke** [comment]
I installed protoc using the instructions [here](https://github.com/google/protobuf/tree/master/src) at a separate time.

By the way, the autogen.sh file has curl to the link https://googlemock.googlecode.com/files/gmock-1.7.0.zip , so it won't work. This file does not exist (Error 404).

**Comment by ojmakke** [comment]
Alright, I think I found the issue.
If you already have protoc installed, its seems that the build will know that and not include any of the header files, and several namespaces will be missing, all of which are in the "third_party/protobuf".

After I "make uninstall"ed my protoc in thirdparty, and attempted to build everything together, I managed to go through the build successfully.

I hope this is helpful information for you. At least there is a way to get the build to work.
